core/src/main/java/org/hisp/dhis/android/core/mockwebserver/Dhis2Dispatcher.java

```diff
@@ -0,0 +1,49 @@
+package org.hisp.dhis.android.core.mockwebserver;
+
+import android.util.Log;
+
+import org.hisp.dhis.android.core.arch.file.IFileReader;
+
+import java.io.IOException;
+
+import okhttp3.mockwebserver.Dispatcher;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.RecordedRequest;
+
+public class Dhis2Dispatcher extends Dispatcher {
+
+    private static final String DISPATCHER = "Dispatcher";
+
+    private IFileReader fileReader;
+    private ResponseController responseController;
+
+    Dhis2Dispatcher(IFileReader fileReader, ResponseController responseController){
+        this.fileReader = fileReader;
```

Comment:
We needed our own implementations of the file reader:
- our jsons sometimes are different than the ones in the sdk/resources
- the current implementation of the IfileReader returns the resources of the sdk
- we work with assets instead